pyqtgraph/functions.py

```diff
@@ -3235,3 +3236,59 @@ def __enter__(self):
     def __exit__(self, *args):
         if self.reconnect:
             self.signal.connect(self.slot)
+
+def connect_lambda(bound_signal, self, func, **kwargs):
```

Comment:
Pulled verbatim from original commit on #1359.
Still need to cherry the last couple commits from that branch that adjusted docs iirc.

Reply:
Actually, so now having tested this a bunch in a parent project, I'm pretty sure it's not needed?

@0000matteo0000 what was the original leak you found which motivated using this?

Reply:
storing self in a lambda or in any object and then storing that object in self (`self.var = {"self": self}`) will break python garbage collection. this is how i test it:
`

import weakref
import gc

import numpy as np
import pyqtgraph as pg
from pyqtgraph.Qt import QtWidgets
app = pg.mkQApp()
pg.setConfigOption('background', 'w')
pg.setConfigOption('foreground', 'k')

mpw = pg.MultiAxisPlotWidget()

# LEGEND
mpw.addLegend(offset=(0, 0))
# TITLE
mpw.setTitle('MultiAxisPlotWidget Example')
# AXYS
mpw.addAxis('sx1', 'bottom', text='Samples1', units='sx1')
mpw.addAxis('sx2', 'bottom', text='Samples2', units='sx2')
mpw.addAxis('sx3', 'bottom', text='Samples3', units='sx3')
mpw.addAxis('sy1', 'left', text='Data1', units='sy1')
mpw.addAxis('sy2', 'left', text='Data2', units='sy2')
mpw.addAxis('sy3', 'left', text='Data3', units='sy3')
# CHARTS
mpw.addChart('Dataset 1', xAxisName='sx1', yAxisName='sy1')[0].setData(np.array(np.sin(np.linspace(0, 2 * np.pi, num=1000))))
mpw.addChart('Dataset 2', xAxisName='sx2', yAxisName='sy1')[0].setData(np.array(np.sin(np.linspace(0, 2 * np.pi, num=1000))) * 2)
mpw.addChart('Dataset 3', xAxisName='sx2', yAxisName='sy2')[0].setData(np.array(np.sin(np.linspace(0, 4 * np.pi, num=500))) * 3)
mpw.addChart('Dataset 4', xAxisName='sx3', yAxisName='sy3')[0].setData(np.array(np.sin(np.linspace(0, 4 * np.pi, num=500))) * 3)
# make and display chart
mpw.makeLayout()
mpw.update()
mpw.show()
pg.exec()

refs = [
    weakref.ref(mpw),
    weakref.ref(mpw.pi),
    weakref.ref(mpw.vb),
    weakref.ref(mpw.layout),
    weakref.ref(mpw.pi.legend),
    weakref.ref(mpw.charts['Dataset 1']),
    weakref.ref(mpw.axes['sx1']),
]
del mpw
for ref in refs:
    print(ref(), 'should be None')
gc.collect()
for ref in refs:
    print(ref(), 'should be None')
`